include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2240,6 +2244,11 @@ class binary_reader
             bool is_error = get_ubjson_size_value(result.first, is_ndarray);
             if (input_format == input_format_t::bjdata && is_ndarray)
             {
+                if (inside_ndarray)
+                {
+                    return sax->parse_error(chars_read, get_token_string(), parse_error::create(112, chars_read,
+                                            exception_message(input_format, "ndarray can not be recursive", "size"), nullptr));
```

Comment:
Please add a test that triggers this exception.

Reply:
done

Reply:
The coverage is still not complete:

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/159488/169705309-a9749510-4388-4bd7-9ed7-2d744aa7d045.png">

(see https://coveralls.io/builds/49342015/source?filename=include%2Fnlohmann%2Fdetail%2Finput%2Fbinary_reader.hpp#L2577)

Reply:
my bad, did not check the report carefully. now it is added.